### PR TITLE
doc: doc fixes to the Hello nRF cloud docs

### PR DESCRIPTION
**Pull request by divipillai**
doc fixes to the Hello nRF cloud docs
as per the review comments recieved.

### CODE REVIEW

File: docs/about.rst [hunk 1]
```diff
@@ -8,9 +8,7 @@ About Thingy:91 X: Hello nRF Cloud firmware
    :depth: 1
 
 This project is based on the `NCS Example Application`_.
-The project contains the firmware that is programmed on the Thingy:91 X in the factory.
-
-.. to do intro
+The project contains the firmware that is programmed on the Thingy:91 X in the factory-programmed.
```
Comment (jtguggedal) [inline docs/about.rst:11]:
```suggestion
The project contains the firmware that is programmed on the Thingy:91 X in the factory.
```

File: docs/architecture.rst [hunk 2]
```diff
@@ -7,7 +7,7 @@ Firmware Architecture
    :local:
    :depth: 2
 
-This is the firmware that is programmed on the nRF9151 of the Thingy:91 X in the factory.
+The firmware on the nRF9151 SiP of the Thingy:91 X is factory-programmed.
```
Comment (jtguggedal) [inline docs/architecture.rst:10]: Is there something missing in this sentence?

File: docs/gsg_guide.rst [hunk 3]
```diff
@@ -54,7 +54,7 @@ Complete the following steps for building and running:
 
       west build -b thingy91x/nrf9151/ns app
 
-#. When using the serial bootloader, you can update using this command:
+#. When using the serial bootloader, you can update the bootloader using the following command:
```
Comment (jtguggedal) [inline docs/gsg_guide.rst:57]:
```suggestion
#. When using the serial bootloader, you can update the application using the following command:
```

File: docs/known_issues.rst [hunk 4]
```diff
@@ -18,3 +18,9 @@ Serial DFU stops working after using GNU screen
   **Workaround:** Use a different serial terminal application, for example, `Serial Terminal app`_, ``tio`` or, power cycle device, if needed.
 
   **Affected platforms:** Thingy:91 X
+This sample requires less hardware resources when it is run on an nRF54 Series device compared to the nRF52 or nRF53 Series devices.
```
Comment (jtguggedal) [inline docs/known_issues.rst:21]: This section is not relevant for this project, as we only support Thingy:91 X here

Comment (divipillai) [reply docs/known_issues.rst]: My bad, this was a copy paste error. The content was of another documentation.

**Pull request merged**